.github/workflows/ruby.yml

```
@@ -1,23 +1,28 @@
 name: Ruby
 
-on: [push]
+on:
+  pull_request: {}
+  push:
+    branches:
+      - main
 
 jobs:
   build:
 
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        ruby: [ '2.5.x', '2.7.x' ]
+        ruby:
+          - 2.5
+          - 2.7
     steps:
     - uses: actions/checkout@master
```

Should we point it to v4 while we're at it?
```suggestion
    - uses: actions/checkout@v4
```